plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/CgroupOOMDetector.java

```diff
@@ -0,0 +1,269 @@
+        if (needStartOOMDetector()) {
+            if (cgroupMount == null) {
+                LOG.warn("System doesn't support OOM events");
+                return;
+            }
+            try {
+                final long memory = dockerConnector.inspectContainer(container).getConfig().getMemory();
+                final OOMDetector oomDetector = new OOMDetector(container, containerLogProcessor, memory);
+                oomDetectors.put(container, oomDetector);
+                oomDetector.start();
+            } catch (IOException e) {
+                LOG.error(e.getLocalizedMessage(), e);
+            }
+        }
+    }
+
+    private boolean needStartOOMDetector() {
+        if (DockerConnector.isUnixSocketUri(dockerDaemonUri)) {
+            return true;
+        }
+        if (SystemInfo.isLinux()) {
+            final String dockerDaemonHost = dockerDaemonUri.getHost();
+            if ("localhost".equals(dockerDaemonHost) || "127.0.0.1".equals(dockerDaemonHost)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /*
+     * Need detect OOM errors and notify users about them. Without such notification if application is killed by oom-killer client often can
+     * see message "Killed" and there is no any why to see why. Unfortunately for now docker doesn't provide clear mechanism how to control
+     * OOM errors, with docker event mechanism can get something like that:
+     * {"status":"die","id":"dfdf82bd3881","from":"base:latest","time":1374067970}
+     * That is not enough.
+     * Found two ways how to control OOM errors.
+     *
+     *     1. With parsing output of 'dmesg' command
+     * ----
+     * andrew@andrey:~> dmesg | grep oom-killer
+     * [41313.629018] java invoked oom-killer: gfp_mask=0xd0, order=0, oom_score_adj=0
+     * [41631.391818] java invoked oom-killer: gfp_mask=0xd0, order=0, oom_score_adj=0
+     * ...
+     * -----
+     * Problem here is in timestamp format. Unfortunately dmesg doesn't provide real time correctly with -T option. Here is a piece of man
+     * page:
+     * -----
+     * -T, --ctime
+     *         Print human readable timestamps.  The timestamp could be inaccurate!
+     *
+     *         The time source used for the logs is not updated after system SUSPEND/RESUME.
+     * -----
+     * So it's complicated to detect time when oom-killer was activated and link its activity with failed docker container.
+     *
+     *     2. Usage of cgroup notification mechanism.
+     * Good article about this: https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/6/html/Resource_Management_Guide/sec-Using_the_Notification_API.html
+     */
+    private static String  cgroupMount;
+    private static boolean systemd;
+
+    static {
+        if (SystemInfo.isLinux()) {
+            final String mounts = "/proc/mounts";
+            try (BufferedReader reader = Files.newBufferedReader(Paths.get(mounts), Charset.forName("UTF-8"))) {
```

Comment:
I would not want to change everything when I just move code from one class to another. It can be done later.